chains/evm/voter/voter.go

```diff
@@ -156,7 +157,27 @@ func (v *EVMVoter) shouldVoteForProposal(prop *proposal.Proposal, tries int) (bo
 		return v.shouldVoteForProposal(prop, tries)
 	}
 
-	return true, nil
+	return v.simulateCall(prop, 0)
+}
+
+func (v *EVMVoter) simulateCall(prop *proposal.Proposal, tries int) (bool, error) {
```

Comment:
It kinda seems to it would be much more simpler to understand if you do this after shouldVoteForProposal instead of integrating it